nerdtree_plugin/vim-nerdtree-tabs.vim

```diff
@@ -463,13 +465,12 @@ endfun
 " s:VimEnterHandler() {{{
 "
 fun! s:VimEnterHandler()
-  " if the argument to vim is a directory, cd into it
-  if g:nerdtree_tabs_startup_cd && isdirectory(argv(0))
-    exe 'cd ' . escape(argv(0), '\ ')
-  endif
+  let l:open_nerd_tree_on_startup = (g:nerdtree_tabs_open_on_console_startup == 1 && !has('gui_running')) ||
+                                  \ (g:nerdtree_tabs_open_on_gui_startup == 1 && has('gui_running'))
 
-  let l:open_nerd_tree_on_startup = (g:nerdtree_tabs_open_on_console_startup && !has('gui_running')) ||
-                                  \ (g:nerdtree_tabs_open_on_gui_startup && has('gui_running'))
+  let l:open_directory_on_startup = isdirectory(argv(0)) && 
+			  \(g:nerdtree_tabs_open_on_console_startup == 2 || 
+			  \ g:nerdtree_tabs_open_on_gui_startup == 2)
```

Comment:
I just tested, the only issue i see is that here when testing for value `2` we should be using the `has('gui_running')` and `!has('gui_running')` conditions to maintain separation between console and gui, as we already are when testing for value `1`. To clarify what i mean, i tested with:

    let g:nerdtree_tabs_open_on_console_startup = 2
    let g:nerdtree_tabs_open_on_gui_startup = 0

and i ran `gvim .` and the program behaved as if the gui option was set to `2` while in fact it was set to `0`.

Otherwise this seemed to work fine and i think it is a neat idea. Thanks!